app/main/posts/views/filters/active-search-filters.directive.js

```diff
@@ -24,19 +35,79 @@ function ActiveSearchFilters($translate, $filter, PostFilters, _, FilterTransfor
+                originalSavedSearch = angular.copy(PostFilters.getModeEntity());
+                $scope.savedSearch = PostFilters.getModeEntity();
+            }
+            /**
+             * This handles the requirement to have saved search filters displayed in a different way
+             * from the rest of the filters.
+             *  - If there is a saved search:
+             *  - - set it in the scope: this is what we will use to display the saved search and its filters in the UI
+             *  - - get clean version (no defaults) of the saved search filters
+             *  - - get a clean currentFilters array that does not include the saved search filters. $scope.uiFilters: this
+             *  is the array we will be using to show the "extra" filters the user can set AFTER they selected  a saved search
+             *  (keep in mind that when a saved search is selected all filters are erased in favor of the saved search)
+             *  - - - Diffing rules: value from currentFilters takes priority over value from search. this is because
+             *  the user will always have selected the value after they select a saved search,meaning they want to change it.
+             *  - - - Diffing rules: if the value is equal, just ignore it/send empty value.
+             *  - - - Diffing rules: return the _difference if the value is not equal, because we will want to show for instance:
+             *  saved search: tag id 1 + filters tag id 2 (so it's not just ignoring the arrays)
+             */
+            $scope.userCanUpdateSavedSearch = false;
+            $scope.uiFilters = currentFilters;
+
+            if ($scope.savedSearch) {
+                /**
+                 * to handle removal correctly we need to make sure we take currentFilters (which has up to date info) into account,
+                 * because that is where our savedsearch filters will stop being represented when we remove them.
+                 * If there's a key in our current filters that is in the saved search but is not in the active filters at this point,
+                 * it is because it was removed (since before saved search gets assigned, they are all assigned to the filters)
+                 * that means we have to remove it from the saved search.
+                 **/
+                $scope.savedSearch.filter = PostFilters.cleanRemovedValuesFromObject(filters, PostFilters.getUIActiveFilters($scope.savedSearch.filter));
+                /**
+                 * Add back in savedSearch.filter if an originally saved search filter is removed+added back
+                 */
+                $scope.savedSearch.filter = PostFilters.getUIActiveFilters(PostFilters.addIfCurrentObjectMatchesOriginal(PostFilters.getUIActiveFilters($scope.savedSearch.filter), PostFilters.getUIActiveFilters(originalSavedSearch.filter), PostFilters.getUIActiveFilters($scope.uiFilters)));
```

Comment:
FYI I moved this to a service and made it more sane too . Pushing today after testing everything again